ui/src/ifql/components/FuncArgs.tsx

```diff
@@ -24,8 +26,32 @@ export default class FuncArgs extends PureComponent<Props> {
       onDeleteFunc,
       declarationID,
       onGenerateScript,
+      declarationsFromBody,
     } = this.props
-
+    const {name: funcName, id: funcID} = func
+    if (funcName === funcNames.JOIN) {
+      return (
+        <div className="func-node--tooltip">
+          <Join
+            func={func}
+            bodyID={bodyID}
+            declarationID={declarationID}
+            onChangeArg={onChangeArg}
+            declarationsFromBody={declarationsFromBody}
+            onGenerateScript={onGenerateScript}
+          />
+          <div className="func-node--buttons">
```

Comment:
This seems to be the same as lines 73 - 82. Perhaps you could make this a constant or its own component to avoid this code being written twice?